spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/config/Neo4jRepositoryConfigurationExtension.java

```diff
@@ -168,35 +185,74 @@ public void registerBeansForRoot(BeanDefinitionRegistry registry, RepositoryConf
 
 		Object source = config.getSource();
 
-		registerIfNotAlreadyRegistered(createSharedSessionCreatorBeanDefinition(config), registry,
-				NEO4J_SHARED_SESSION_CREATOR_BEAN_NAME, source);
+		this.sessionBeanName = registerWithSuggestedNameOrGenerateNewName(createSharedSessionCreatorBeanDefinition(config),
```

Comment:
Bean registration should either use generated names or provided ones, see also for the mapping context. Mixing behavior can be confusing.